### PR TITLE
Store insecure in the context and use automatically if present

### LINKED ISSUE #63: Zed context should store whether the context is insecure
Right now the context stores everything I need to talk to Authzed.com or a local SpiceDB except whether that connection should use TLS or not. This means that every zed request to SpiceDB needs the `--insecure` flag. This flag should be stored along with the rest of the context because it doesn't regularly change for a context.

### PR DESCRIPTION
**Pull request by josephschorr**
Fixes #63

### CODE REVIEW

File: cmd/zed/context.go [hunk 1]
```diff
@@ -77,18 +77,33 @@ func contextListCmdFunc(cmd *cobra.Command, args []string) error {
...
+				secret = "<redacted>"
+			} else {
+				secret = stringz.Join("_", prefix, "<redacted>")
+			}
+		}
+
+		insecure := false
+		if token.Insecure != nil {
+			insecure = *(token.Insecure)
+		}
```
Comment (jzelinskie) [inline cmd/zed/context.go:90]: Also might want to make this a method on the token

File: cmd/zed/context.go [hunk 2]
```diff
@@ -77,18 +77,33 @@ func contextListCmdFunc(cmd *cobra.Command, args []string) error {
...
+
+		insecureStr := ""
+		if insecure {
+			insecureStr = "   ✓    "
```
Comment (jzelinskie) [inline cmd/zed/context.go:94]: We should make this a constant now that two things are using it

File: cmd/zed/main.go [hunk 3]
```diff
@@ -36,16 +36,16 @@ func defaultStorage() (storage.ConfigStore, storage.SecretStore) {
...
 }
 
-func dialOptsFromFlags(cmd *cobra.Command, token string) []grpc.DialOption {
+func dialOptsFromFlags(cmd *cobra.Command, token storage.Token) []grpc.DialOption {
 	opts := []grpc.DialOption{
 		grpc.WithUnaryInterceptor(zgrpcutil.LogDispatchTrailers),
 	}
 
-	if cobrautil.MustGetBool(cmd, "insecure") {
+	if cobrautil.MustGetBool(cmd, "insecure") || (token.Insecure != nil && *token.Insecure) {
```
Comment (jzelinskie) [inline cmd/zed/main.go:44]: if it was a method this would just be `|| token.Insecure()`

**Review by jzelinskie — CHANGES_REQUESTED** [review]

File: cmd/zed/context.go (hunk 2 above)
Comment (josephschorr) [reply cmd/zed/context.go:94]: They are actually slightly different in length

**Comment by josephschorr** [comment]
Updated

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**